### DISCUSSION TITLE
[QUESTION] How to have shared info cached for multiple commands

### DISCUSSION THREAD

**Discussion by shelper** [fastapi/typer#1312] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the Typer documentation, with the integrated search.
* [x] I already searched in Google "How to X in Typer" and didn't find any information.
* [x] I already searched in Google "How to X in Click" and didn't find any information.

### Description
say i have a script like below with two typer commands, where I want to load certain info in one command and then use the info in another command. the code below won't work but it explains what i want, and i wonder if it is possible to do this using typer.

```
import typer
app = typer.Typer()

__input__ = ''

@app.command()
def load(input:str):
    __input__ = input

    
@app.command()
def display():
    print(__input__)

if __name__ == "__main__":
    app()
```

then in my command line window, i can do:
```
python main.py load abc
python main.py display # will print out 'abc'
```

**Comment by sathoune** [comment, marked as answer]
You need some kind of persistence. After you run command and it finishes work, program is closed so no cache will be available for the second command. You can write your results to some file or you could use sqlite3 within python.

**Comment by YuriiMotov** [comment]
See also: #185
